SpatialGDK/Documentation/content/non-unreal-server-worker-types.md

```diff
@@ -82,19 +104,69 @@ type UnrealResponse {
     string some_response_string = 1;
 }
 
-component NonUnrealAuthoritative {
+component TestComponent {
     id = 1337;
     uint32 counter = 1;
+    command UnrealResponse test_command(UnrealRequest);
 }
 
-component UnrealAuthoritative {
+component OtherTestComponent {
     id = 1338;
     uint32 other_counter = 1;
-    command UnrealResponse test_command(UnrealRequest);
 }
 ```
 
-#### Send data
+#### Using the code generator example
+
+Using the above schema file placed in the project relative directory `spatial\schema\improbable\`, the command below will output code to a new `Game\Source\<your_project_name>\ExternalSchemaCodegen` folder:
+
+`Game\Plugins\UnrealGDK\SpatialGDK\Build\Scripts\ExternalSchemaCodegen.bat spatial\schema\my_external_schema Game\Source\<your_project_name>\ExternalSchemaCodegen`
+
+This will generate code in the `Game\Source\<your_project_name>\ExternalSchemaCode` folder. To see these changes, the Visual Studio solution needs to be regenerated.
+
+To send or receive network operations, you need to instantiate the `ExternalSchemaInterface` class inside your game module.
+
+To send component updates or command requests / responses, you can call any of the overrides for `SendComponentUpdate`, `SendCommandRequest(...)` or `SendCommandResponse(...)`, for example:
+
+```
+auto Update = new ::improbable::testing::TestComponent::Update(20 /* counter field */);
+ExternalSchema->SendComponentUpdate(30 /* entity_id */, Update);
+```
+
```

Comment:
In addition, there seems to be generated functions for each field (SendCounterUpdate()) etc. I think we should mention at least the existence of these helper functions.

Reply:
The generated fields for each field are constructor pattern style methods, added a comment between these two lines explaining this:

`// further edit Update using constructor pattern methods, e.g. (void)Update->SetCounter(30);`